src/client.rs

```diff
@@ -499,6 +502,100 @@ impl<T: Read + Write> Client<T> {
     }
 }
 
+#[cfg(feature = "rsasl")]
+impl<T: Read + Write> Client<T> {
+
+    /// Authenticate with the server using the given custom SASLConfig to handle the server's
+    /// challenge.
+    ///
+    pub fn sasl_auth(
```

Comment:
How do you feel about also adding `#[deprecated]` to `fn authenticate` and point people at this instead?

Reply:
I think that's a good idea ^^